### DISCUSSION TITLE
Table `migrations` missing column `id`

### DISCUSSION THREAD

**Discussion by MircoBabin** [laravel/framework#39473] [Q&A] [answered]
Dear reader,

Today I noticed that older installations (originating from Laravel 4.2) are missing the `id` column in the `migrations` table. Newer installations do have the `id` column.

I found the creation of the `migrations` table in [laravel\framework\src\Illuminate\Database\Migrations\DatabaseMigrationRepository.php](https://github.com/laravel/framework/blob/f429789d8c22cc51f0712d708db00d8117e9ce0e/src/Illuminate/Database/Migrations/DatabaseMigrationRepository.php#L146) `function createRepository()`.

The `id` column is added in pull request #15770 (6 oct 2016, Laravel 5.3). In this [unanswered comment](https://github.com/laravel/framework/pull/15770#issuecomment-252064043) is asked what existing installations should do.

I have read the upgrade notes for Laravel 5.3, but there is no mention of a change in the `migrations` table.

**Does anybody with an elephant history recall if any migration exists or is documented somewhere how to add the `id` column if it does not exists to the `migrations` table?**

P.S. this project was shifted to Laravel 7.x (current version) and will shift to Laravel 9.x (LTS) when it comes available.

Kind Regards,
Mirco Babin

**Comment by dennisprudlo** [comment]
This shouldn't be a problem, the `id`-column isn't used. It was just added to satisfy the database. We have no IDs in our `migrations`-table with Laravel 8 on our staging environment. But our local environments `migration`-table have an `id` column because the databases there were re-created sometime after the reference PR.

If it really bugs you, you can manually add the `id`-column. As it says in the docs (https://laravel.com/docs/8.x/migrations#column-method-increments) it's just an auto-increment UNSIGNED INT.

**Comment by MircoBabin** [comment, marked as answer]
Thanks for your reply @dennisprudlo .

It bugs me, I want all databases to have same table layout. So I created a migration that at least works on MySql.

2021_11_15_070000_update_migrations_id.php
```php
<?php

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

class UpdateMigrationsId extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        if (Schema::hasColumn('migrations', 'id')) {
            return;
        }
        
        //create `migrations`.`id` as first field,  before `migration`
        Schema::table('migrations', function ($table) {
            $table->increments('id')->first();
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        if (Schema::hasColumn('migrations', 'id')) {
            Schema::table('migrations', function ($table) {
                $table->dropColumn('id');
            });
        }
    }
}

```
